ARE/middleware/src/main/java/eu/asterics/mw/are/BundleManager.java

```diff
@@ -132,16 +144,173 @@ public BundleManager(ModelValidator modelValidator) {
+        	        generateCacheFiles();
+        	    }
+        	    storeMD5Sum(md5Sum);
+        	}
+            initComponentTypeIDToJarNameMap();
+            logger.fine("BundleManager initialization finished");
+        } catch (IOException | ParseException | URISyntaxException | NoSuchAlgorithmException e1) {
             // TODO Auto-generated catch block
             AstericsErrorHandling.instance.reportError(null,
                     "Could not create cache for Asterics components:\n" + e1.getMessage());
         }
+    }
+    
+    /**
+     * Initializes the internal maps either with caching (in OSGI mode) or without caching in NON-OSGI mode (used by APE).
+     * @throws IOException
+     * @throws URISyntaxException
+     * @throws ParseException
+     */
+    private void initComponentTypeIDToJarNameMap() throws IOException, URISyntaxException, ParseException {
+        if(ResourceRegistry.getInstance().isOSGIMode()) {
+            readComponentListCache();
+        } else {
+            logger.fine("Not in OSGI Mode --> init maps without cache files.");
+            componentTypeIDToJarName.clear();
+            List<URI> jarNameURIList=ResourceRegistry.getInstance().getComponentJarList(false);
+            for(URI jarNameURI : jarNameURIList) {
+                URI bundleDescriptorURI=ResourceRegistry.toJarInternalURI(jarNameURI, DefaultBundleModelParser.BUNDLE_DESCRIPTOR_RELATIVE_URI);
+                String bundleDescriptorAsXMLString=ResourceRegistry.getResourceContentAsString(bundleDescriptorURI.toURL().openStream());
+                Set<IComponentType> componentTypeSet = DefaultBundleModelParser.instance.parseModelAsXMLString(bundleDescriptorAsXMLString);
+                for(IComponentType componentType : componentTypeSet) {
+                    componentTypeIDToJarName.put(componentType.getID(),ResourceRegistry.toString(ResourceRegistry.getInstance().toRelative(jarNameURI)));
+                    ComponentRepository.instance.install(componentType);
+                }
+            }
+            logger.fine("Found the following componentTypeIds: "+componentTypeIDToJarName.keySet().toString());
+        }
+    }
 
-        installServices();
+    /**
+     * Checks one of the used cache files is misssing.
+     * @return
+     */
+    private boolean cacheFileMissing() {
+        logger.fine("Checking missing cache files");
+        if(!ResourceRegistry.resourceExists(LOADER_COMPONENTLIST_CACHE_FILE_URI) ||
+                !ResourceRegistry.resourceExists(COMPONENT_COLLECTION_CACHE_FILE_URI)) {
+                return true;
+        }
+        return false;
+    }
+
+    /**
+     * Compares the given md5sum hex string and checks it for equality with a newly created md5 hash string of all plugin jars.
+     * @param md5Sum
+     * @return
+     * @throws URISyntaxException
+     * @throws IOException
+     */
+    private boolean bundleChangeDetected(String md5Sum) throws URISyntaxException, IOException {
```

Comment:
shouldnt `URISyntaxException` and `IOException` be catched and 'true' returned in this case?